Backend/Remora.Discord.Caching/CacheResult.cs

```diff
@@ -0,0 +1,64 @@
+//
+//  CacheResult.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) 2017 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Remora.Discord.Caching
+{
+    /// <summary>
+    /// Result from the cache.
+    /// </summary>
+    /// <typeparam name="T">Type of the value.</typeparam>
+    public readonly struct CacheResult<T> where T : notnull
```

Comment:
This type is redundant; everything it does is available through `Result<T>`. Use that type instead, if there are no particular reasons for adding this type.